File: python/distro.py
#!/usr/bin/python

# Report the distribution we are running on.

import sys, subprocess, re

issue = subprocess.check_output(['cat', '/etc/issue'])

if re.search('openSUSE',issue,re.IGNORECASE) != None:
	print "openSUSE\n"
elif re.search('fedora',issue,re.IGNORECASE) != None:
	print "fedora\n"
elif re.search('ubuntu',issue,re.IGNORECASE) != None:
	print "ubuntu\n"
else:
	print "unknown\n"

sys.exit(0)

File: perl/distro.pl
#!/usr/bin/perl

#Report the distribution we are running on.

use strict;

my $issue = `cat /etc/issue`;

if ($issue =~ /openSUSE/i) {
	print "openSUSE\n";
}
elsif ($issue =~ /fedora/i) {
	print "fedora\n";
}
elsif ($issue =~ /ubuntu/i) {
	print "ubuntu\n";
}
else {
	print "unknown\n";
}

exit(0);

